File: my-folder/0020-valid-parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        n = len(s)
        

        stack = []

        for char in s:
            if char in ['(', '{', '[']:
                stack.append(char)
            else:
                if len(stack) > 0:
                    if char == ')' and stack[-1]!= '(':
                        return False
                    elif char == '}' and stack[-1]!='{': return False
                    elif len(stack)>0 and char == ']' and stack[-1]!='[': return False
                    else:
                        stack.pop(-1)
                else:
                    return False
        return len(stack)==0

File: my-folder/0020-valid-parentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;

        for (auto ch: s){
            if (ch == ')'){
                if (stack.size() < 1 || stack.top() != '('){
                    return false;
                }
                else{
                    stack.pop();
                }
            }

            else if (ch == '}'){
                if (stack.size() < 1 || stack.top() != '{'){
                    return false;
                }
                else{
                    stack.pop();
                }
            }

            else if (ch == ']'){
                if (stack.size() < 1 || stack.top() != '['){
                    return false;
                }
                else{
                    stack.pop();
                }
            }
            else
                stack.push(ch);
        }

        if (stack.size() == 0)
            return true;
        else
            return false;
        
    }
};
